kubernetes/doc/edge-deployment-to-k8s-translations.md

```diff
@@ -0,0 +1,215 @@
+ContainerCreate structure](https://docs.docker.com/engine/api/v1.40/#operation/ContainerCreate). 
+This means Edge on K8s will have to make decisions about how to provide the module framework based 
+on these settings. This document will describe which values are used from the module specification 
+and how they are transformed into Kubernetes objects.
+
+The application will create Deployments, Services, ImagePullSecrets, PersistentVolumeClaims, and 
+ServiceAccounts to establish this framework.
+Each will be described in detail in subsequent sections.
+
+## Deployment
+
+Each Edge Module will create one Deployment. 
+
+There will be a default label set used throughout the deployment. Three labels assigned to every 
+object, all values will be sanitized to be a K8S label value:
+- **net.azure-devices.edge.module** = module name
+- **net.azure-devices.edge.deviceid** = device id
+- **net.azure-devices.edge.hub** = associated IoT Hub name.
+
+### metadata
+
+- **name**        = Name will be the module name, sanitized to be a K8s identifier.
+- **labels**      = Default label set
+- **annotations** = Deployments will have the original JSON creation string added in the field "net.azure-devices.edge.creationstring".
+
+### spec (DeploymentSpec)
+
+- **replicas** = Currently set to 1 - only 1 Pod will be created for each module.
+- **selector** = This currently matches the Default label set.
+- **template** = See [PodTemplate](#podtemplate)
+
+### PodTemplate
+
+##### metadata
+
+- **name**        = Name will be the module name, sanitized to be a K8s identifier.
+- **labels**      = Default label set
+- **annotations** = The pod's metadata will have one fixed annotation:
+    - **net.azure-devices.edge.original-moduleid** = unsanitized module id from Edge Deployment Spec.
+    - then `settings.createOptions.Labels` will be added to the pod's annotations.
+    
+##### spec (PodSpec)
+- **containers**
+    - **Proxy container**
+      This will contain the specification for the module to iotedged proxy.
+    - **Module container**
+        - **name** = Name will be the module name, sanitized to be a K8s identifier.
+        - **image** = `settings.image`
+        - **env** = env will contain some predefined settings for all modules, then:
+            - Add all environment variables from `env` section in module spec
+            - Add all environment variables from `settings.createOptions.Env`
+        - **securityContext**
+            - privileged = Derived from `settings.createOptions.HostConfig.Privileged`
+        - **volumeMounts** = There are 4 sources to volume mounts.
+            - bind mounts from `settings.createOptions.HostConfig.Binds` in format "host path:target path[:readwrite mode]".
+              *This is not recommended for Edge on K8s.*
+                - Name = Sanitized host path
+                - MountPath = target path
+                - readOnlyProperty = readwrite mode if set, default is false
```

Comment:
Looking at these defaults - we should have a separate README.md that can be grepped by helm describe a bit better. Take a look at this stable chart as a good example - https://github.com/helm/charts/tree/master/stable/dex

Reply:
We need that, especially for the Helm chart, but this document is trying to be as comprehensive as I can, and this section in particular doesn't organize well into a table.